### ISSUE TITLE
Flake8 Fixes

### ISSUE THREAD

**Issue by jbellister-slac** [slaclab/pydm#855]
There are currently about 700 warnings generated by Flake8. The vast majority are very simple fixes. Should wait for the tail end of the release cycle to do this so it doesn't impact all PRs currently in progress, but would be nice to clean up soon.

**Comment by nstelter-slac** [comment]
any thoughts on also using black (https://github.com/psf/black)?

easy way ensure good formatting, and can be run in addition to Flake8 so we still get linter checks but also get automatic formatting.

**Comment by nstelter-slac** [comment]
ran the following tools on pydm/widgets folder as a test:
-"black --line-length 120 pydm/widgets/"
- "isort  --profile black" 
-"flake8 --max-line-length 120 --extend-ignore=E203 pydm/widgets"

https://github.com/nstelter-slac/pydm/tree/formatting_test

**Comment by klauer** [comment]
[`ruff`](https://github.com/astral-sh/ruff) is also rather nice as it handles both linting and fixing a bunch of different things, from import sorting (`isort`), code formatting (`black`), outdated Python syntax (`pyupgrade`), and a lot more.

**Comment by nstelter-slac** [comment]
hi

oh nice, thanks for the info!
looks good to me, seems like ruff is intended to be used alongside black (https://beta.ruff.rs/docs/faq/#is-ruff-compatible-with-black), so black -> ruff looks to gives us some extra functionality beyond flake8 and also saves using a 3rd tool.

 can try out black -> ruff on some files and see how it goes.

**Comment by nstelter-slac** [comment]
applied black formatting and then ruff linting on the widgets folder, here: https://github.com/slaclab/pydm/compare/master...nstelter-slac:pydm:black_formatting_and_ruff_linting

what i did:
_black --line_length 120 .
ruff .
ruff . --fix
(manually fixed few remaining issues)_

"ruff . --fix" is nice feature, but couldn't fix all the linting errors and can be a bit destructive if not careful (cleared out __init__.py for unused imports, before i set it to ignore the file)

also ruff seems much faster then flake8!
-flake8  2.461 seconds
-ruff      0.013 seconds

**Comment by klauer** [comment]
Yeah, it is quite a bit faster. Looks like you'll be a top PyDM contributor after that PR 😁 

Those `__init__.py` imports are _technically_ unused and probably belong either in `__all__` or use the "public export notation" that py.typed/pyright uses - `from .x import y as y`. For example, `__init__.py` it should read something like `from .application import PyDMApplication as PyDMApplication` to indicate it's public ([ref](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface))
